src/cdk/schematics/ng-update/update-schematic.md

```diff
@@ -20,32 +19,57 @@ given target version:
+changes of the target version. Most of the time we can apply such automatic migrations, but
 there are also a few breaking changes that cannot be migrated automatically.
   
 In that case, our goal should be to notify the developer about the breaking change that needs
-developer attention.  
+attention.
   
 ## Transforming TypeScript files  
   
-In order to automatically migrate TypeScript source files, we take advantage of the `tslint`
-which allows us to create custom rules that can:
- 
-* Easily `visit` specific types of TypeScript nodes (e.g. `visitClassDeclaration`)  
-* Structure migrations based on the _upgrade data_ or type of migration (different TSLint rules)  
-* Easily apply replacements / fixes for specific TypeScript nodes.  
-* Easily report breaking changes at TypeScript nodes that cannot be migrated automatically  
-* Double check for rule migrations (TSLint always runs rule again after migrations have been applied)  
+In order to automatically migrate TypeScript source files, we take advantage of the TypeScript
+Compiler API which allows us to parse and work with the AST of project source files. We built
+a small framework for analyzing and updating project source files that is called `update-tool`.
+
+The `update-tool` has been specifically built with the goal of being extremely fast and
+flexible. This tool had to be built because our initial `ng update` implementation which
+leveraged `tslint` caused various problems:
+
+* No support for HTML templates and stylesheets (workaround was needed)
+* Reruns all upgrade lint rules after file has been updated (significant performance issue for projects with a lot of files)
+* Recreates the TypeScript program each time a source file has been updated (significant memory pressure for big TypeScript projects, causing OOM exceptions)
+* TSLint recursively visits the nodes of all source files for each upgrade lint rule (performance issue)
+* TSLint is not guaranteed to be installed in CLI projects. See: https://github.com/angular/angular-cli/issues/14555
+* TSLint replacements lead to memory leaks due to the retained TypeScript nodes
+* No way to have a *global analysis* phase since lint rules are only able to visit source files.
+* No flexibility. i.e. 
+  * No way to ensure source files are only analyzed a single time 
+  * No way to implement a progress bar
+  * No easy way to add support for HTML templates or stylesheets
+
+All of these problems that `tslint` had, have been solved when we built the
+`update-tool`. The tool currently has the following differences compared to `tslint`:
+
+* Integrated support for the HTML templates and stylesheets
+* Only runs migration rules once per source file.
+    * Even if a source file is part of multiple TypeScript projects.
+* Program is only created once per TypeScript project. Type checker is only create
+* Migration failures are guaranteed to not retain `ts.Node` instances.
```

Comment:
Maybe just make it slightly more clear that this means not leaking memory?

Reply:
Updated!